src/Tes/Repository/TesTaskPostgreSqlRepository.cs

```diff
@@ -0,0 +1,169 @@
+        /// <returns></returns>
+        public async Task<TesTask> CreateItemAsync(TesTask item)
+        {
+            using var dbContext = createDbContext();
+            var dbItem = new TesTaskDatabaseItem { Json = item };
+            dbContext.TesTasks.Add(dbItem);
+            await dbContext.SaveChangesAsync();
+            return item;
+        }
+
+        /// <summary>
+        /// Encapsulates TesTasks as JSON
+        /// </summary>
+        /// <param name="item">TesTask to store as JSON in the database</param>
+        /// <returns></returns>
+        public async Task<List<TesTask>> CreateItemsAsync(List<TesTask> items)
+        {
+            using var dbContext = createDbContext();
+
+            foreach (var item in items) 
+            {
+                var dbItem = new TesTaskDatabaseItem { Json = item };
+                dbContext.TesTasks.Add(dbItem);
+            }
+
+            await dbContext.SaveChangesAsync();
+            return items;
+        }
+
+        /// <summary>
+        /// Base class searches within model, this method searches within the JSON
+        /// </summary>
+        /// <param name="tesTask"></param>
+        /// <returns></returns>
+        public async Task<TesTask> UpdateItemAsync(TesTask tesTask)
+        {
+            using var dbContext = createDbContext();
+            var item = await dbContext.TesTasks.FirstOrDefaultAsync(t => t.Json.Id == tesTask.Id);
+
+            if (item is null)
+            {
+                throw new Exception($"No TesTask with ID {tesTask.Id} found in the database.");
+            }
+
+            item.Json = tesTask;
+            await dbContext.SaveChangesAsync();
+            return item.Json;
+        }
+
+        /// <summary>
+        /// Base class deletes by Item.Id, this method deletes by Item.Json.Id
+        /// </summary>
+        /// <param name="id">TesTask Id</param>
+        /// <returns></returns>
+        public async Task DeleteItemAsync(string id)
+        {
+            using var dbContext = createDbContext();
+            var item = await dbContext.TesTasks.FirstOrDefaultAsync(t => t.Json.Id == id);
+
+            if (item is null)
+            {
+                throw new Exception($"No TesTask with ID {item.Id} found in the database.");
+            }
+
+            dbContext.TesTasks.Remove(item);
+            await dbContext.SaveChangesAsync();
+        }
+
+        /// <summary>
+        /// Identical to GetItemsAsync, paging is not supported. All items are returned
+        /// </summary>
+        /// <param name="predicate">Predicate to query the TesTasks</param>
+        /// <param name="pageSize">Ignored and has no effect</param>
+        /// <param name="continuationToken">Ignored and has no effect</param>
+        /// <returns></returns>
+        public async Task<(string, IEnumerable<TesTask>)> GetItemsAsync(Expression<Func<TesTask, bool>> predicate, int pageSize, string continuationToken)
```

Comment:
I think is cleaner to make a concrete return type with these two values - a record. Alternatively, you could rename the method GetAllItemsAsync, and just return IEnumerable, and implement pagination later, as another method.

Reply:
This method signature comes from the original interface for Cosmos DB; will create a separate issue to harmonize this once Cosmos DB completely removed